### PR TITLE
Moves most punch damage modifiers to physiology + bloodsucker punch damage tweaks

### PR DESCRIPTION
**Pull request by Mqiib**
Since species changing completely fucks up punch damage modifier stuff and resets them and then you get boned after the bonuses are removed I added an extra bit to the pysiology datum so that THAT bit can be changed independant of your species and you should retain your increased punchies. I added three new procs that just return the total physiology + species punch damage values, one each for the max damage, min damage, and stun threshold.

Also messes around with bloodsucker punch stuff because it was all over the goddamn place and didn't have stun threshold stuff so you could just get soap bar hands in certain situations. As it turns out lasombra clan could also get infinite punch damage for the low low cost of spamming a 30-blood-cost ability (fortitude) after ascending it and ranking it up once or twice, netting them anywhere from 0.25 to like 1.3 damage per use permanently with each use because apparently someone didn't rtealize that a*b and a/b can be non-equivalent values. PRETTY SURE it all works because it compiles and I ran through the math like at least four times in my head. Maybe three. I don't remember any more, it's 2 am.

Please tell me if I missed any mentions of punch damage. I know I specifically skipped over some bloodsucker ones because they mess with species as well and didn't feel like buggering around with that.

# Wiki Documentation

I dunno if there's any mention of the physiology datum but it includes punch damage bonuses now. This shouldn't ACTUALLY affect anything mechanically besides making bloodsuckers not so prone to slapping people to the ground with every swing.

# Changelog

:cl:  
rscadd: Adds punch damage mods to the physiology datum
rscadd: Also adds three new human procs to get the total punch damage values
bugfix: Makes punch damage modifiers more consistent after species changes
bugfix: Bloodsucker punch damage (re)calculations changed to be more consistent
bugfix: Lasombra clan can no longer permanently trade 30 blood for more punch damage infinitely
tweak: Bloodsuckers no longer get insane knockdown chance on hit
tweak: Martial arts now use the new procs instead of just species damage
/:cl:

### CODE REVIEW

**Comment by Mqiib** [comment]
Ok I have tested it and it seems to work fine so we're good, probably!

**Review by ToasterBiome — APPROVED** [review]

**Comment by ToasterBiome** [comment]
You change bloodsuccker stuff so this is balance stuf............... sorry.................

**Review by JamieD1 — APPROVED** [review]

File: code/datums/mutations/body.dm [hunk 1]
```diff
@@ -280,19 +280,20 @@
...
 		return
 	var/strength_punchpower = GET_MUTATION_POWER(src) * 2 - 1 //Normally +1, strength chromosome increases it to +2
-	owner.dna.species.punchdamagelow += strength_punchpower
-	owner.dna.species.punchdamagehigh += strength_punchpower
-	owner.dna.species.punchstunthreshold += strength_punchpower //So we dont change the stun chance
-	ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
+	owner.physiology.punchdamagehigh_bonus += strength_punchpower
+	owner.physiology.punchdamagelow_bonus += strength_punchpower
+	owner.physiology.punchstunthreshold_bonus += strength_punchpower //So we dont change the stun chance
+  ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)
```
Comment (JamieD1) [inline code/datums/mutations/body.dm]:
```suggestion
		ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
```

File: code/datums/mutations/body.dm [hunk 2]
```diff
@@ -280,19 +280,20 @@
...
 		return
 	var/strength_punchpower = GET_MUTATION_POWER(src) * 2 - 1
-	owner.dna.species.punchdamagelow -= strength_punchpower
-	owner.dna.species.punchdamagehigh -= strength_punchpower
-	owner.dna.species.punchstunthreshold -= strength_punchpower
-	REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
+	owner.physiology.punchdamagehigh_bonus -= strength_punchpower
+	owner.physiology.punchdamagelow_bonus -= strength_punchpower
+	owner.physiology.punchstunthreshold_bonus -= strength_punchpower
+  REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)
```
Comment (JamieD1) [inline code/datums/mutations/body.dm]:
```suggestion
		REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]

File: code/datums/mutations/body.dm (hunk 1 above)
Comment (JamieD1) [inline code/datums/mutations/body.dm]:
```suggestion
		ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
```

File: code/datums/mutations/body.dm [hunk 3]
```diff
@@ -280,19 +280,20 @@
...
```
Comment (JamieD1) [inline code/datums/mutations/body.dm]:
```suggestion
		REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
```

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 4]
```diff
@@ -277,3 +277,13 @@
 
 /mob/living/carbon/human/get_biological_state()
 	return dna.species.get_biological_state()
+
```
Comment (JamieD1) [inline code/modules/mob/living/carbon/human/human_helpers.dm:280]:
```suggestion
```

File: code/datums/mutations/body.dm [hunk 5]
```diff
@@ -280,19 +280,20 @@
...
 		return
 	var/strength_punchpower = GET_MUTATION_POWER(src) * 2 - 1
-	owner.dna.species.punchdamagelow -= strength_punchpower
-	owner.dna.species.punchdamagehigh -= strength_punchpower
-	owner.dna.species.punchstunthreshold -= strength_punchpower
-	REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
+	owner.physiology.punchdamagehigh_bonus -= strength_punchpower
+	owner.physiology.punchdamagelow_bonus -= strength_punchpower
+	owner.physiology.punchstunthreshold_bonus -= strength_punchpower
+		REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)
```
Comment (JamieD1) [inline code/datums/mutations/body.dm]:
```suggestion
	REMOVE_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
```

File: code/datums/mutations/body.dm [hunk 6]
```diff
@@ -280,19 +280,20 @@
...
 		return
 	var/strength_punchpower = GET_MUTATION_POWER(src) * 2 - 1 //Normally +1, strength chromosome increases it to +2
-	owner.dna.species.punchdamagelow += strength_punchpower
-	owner.dna.species.punchdamagehigh += strength_punchpower
-	owner.dna.species.punchstunthreshold += strength_punchpower //So we dont change the stun chance
-	ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
+	owner.physiology.punchdamagehigh_bonus += strength_punchpower
+	owner.physiology.punchdamagelow_bonus += strength_punchpower
+	owner.physiology.punchstunthreshold_bonus += strength_punchpower //So we dont change the stun chance
+		ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)
```
Comment (JamieD1) [inline code/datums/mutations/body.dm]:
```suggestion
	ADD_TRAIT(owner, TRAIT_QUICKER_CARRY, src)	
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**